### PR TITLE
Upgrade embedded Kotlin to 2.0.20

### PR DESCRIPTION
**Pull request by eskatos**
* Follows up https://github.com/gradle/gradle/pull/30127
* See https://github.com/JetBrains/kotlin/releases/tag/v2.0.20
* Wrapper update is necessary for bumping the `kotlin-dsl` plugin version and the associated Kotlin version.
* Another wrapper update will be necessary once this is merged to get back to a regular nightly.

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by eskatos** [comment]
@bot-gradle test RFN

**Comment by eskatos** [comment]
@bot-gradle test RFN

**Comment by eskatos** [comment]
@bot-gradle test RFN

**Comment by eskatos** [comment]
@bot-gradle test RFN

File: testing/distributions-integ-tests/src/integTest/groovy/org/gradle/DistributionIntegrationSpec.groovy [hunk 1]
```diff
@@ -34,7 +34,7 @@ import static org.hamcrest.MatcherAssert.assertThat
 
 abstract class DistributionIntegrationSpec extends AbstractIntegrationSpec {
 
-    protected static final THIRD_PARTY_LIB_COUNT = 144
+    protected static final THIRD_PARTY_LIB_COUNT = 145
```
Comment (h0tk3y) [inline testing/distributions-integ-tests/src/integTest/groovy/org/gradle/DistributionIntegrationSpec.groovy:37]: Does the new library come as a transitive dependency on the new Kotlin version? I did not notice new dependencies in this PR.

**Review by h0tk3y — APPROVED** [review]
LGTM, but I was not familiar with the regular procedure of the Kotlin version upgrade.

File: testing/distributions-integ-tests/src/integTest/groovy/org/gradle/DistributionIntegrationSpec.groovy (hunk 1 above)
Comment (eskatos) [reply testing/distributions-integ-tests/src/integTest/groovy/org/gradle/DistributionIntegrationSpec.groovy:37]: Yes, one jar got split in the Kotlin transitives.

**Pull request merged**